src/api/metadataApi.ts

```diff
@@ -12,6 +12,7 @@ import { METADATA_RETRY_LIMIT, DOWNLOAD_METADATA_RETRY_INTERVAL } from '../const
 import { GetDagMetadataApiError } from '../errors';
 
 const metadataApi = {
+  // eslint-disable-next-line consistent-return -- The return is consistent, but too complex for the rule
```

Comment:
Instead of ignoring the rule maybe we could have a type for the return of this method and `return response.data as newType`.

This would make callers aware of the type and would solve the issue with the rule

Reply:
Actually, the problem with this rule wasn't the type, but the fact that some code branches would never explicitly `return`, when others did.

To fix this, a more intrusive refactor was necessary with potential to add bugs that would be hard to fetch. Since this method had no unit tests I decided to solve this in another PR:
- #702 

Would you agree in leaving this `eslint-disable` here and solving it through this other PR?